.github/workflows/armada-release.yml

```diff
@@ -0,0 +1,88 @@
+name: Release Armada
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  release-armadactl:
+    runs-on: buildjet-4vcpu-ubuntu-2204
+    strategy:
+      matrix:
+        # We obviously will not matrix releases; but
+        # the go-setup workflow expects this to be set
+        # via a matrix
+        go: [ '1.18' ]
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/workflows/go-setup
+      - run: make build-armadactl-release
+      - name: Upload binaries to release
+        uses: svenstaro/upload-release-action@v2
+        with:
+          file: dist/*
+          file_glob: true
+      - name: Upload artifact to github actions
+        uses: actions/upload-artifact@v3
+        with:
+          name: armadactl-release
+          path: dist/*
+
+  release-docker-images:
+    runs-on: buildjet-4vcpu-ubuntu-2204
+    strategy:
+      matrix:
+        # We obviously will not matrix releases; but
+        # the go-setup workflow expects this to be set
+        # via a matrix
+        go: [ '1.18' ]
+    steps:
+      - name: Pull and re-tag docker images
+        run: |
+          # TODO(JayF): Need to ensure we have a job that uploads docker images on push to master
+          # TODO(JayF): how to get these secrets in here safely?
```

Comment:
Can't you just add a secret to GitHub?  That seems to be the most secure option?